### PR TITLE
Exercise 3

### PR DESCRIPTION
**Pull request by maheshpatelhome**

### CODE REVIEW

File: todo_app/tests/todo_view_model_test.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
+from todo_app.todo_view_model import ToDoViewModel
+from todo_app.trello_card  import TrelloCard
+import pytest
+from datetime import date, datetime, timedelta
+
+@pytest.fixture
+def items_setup():
+    items = []
```
Comment (Jonesey13) [inline todo_app/tests/todo_view_model_test.py]: I'd just create these as one big array (avoiding the appends later) e.g. 
```python
[
    TrelloCard("To Do", 1, 1, "To Do Item 1", "2020-12-01T00:00:00.000Z"),
    TrelloCard("To Do", 1, 2, "To Do Item 2", "2020-12-01T00:00:00.000Z" )
]
```

File: todo_app/todo_view_model.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
+class ToDoViewModel:
+    def __init__(self, items, recent_date):
+        self._items = items
+        self.determine_done_items() 
+        self.determine_recent_done_items(recent_date)
+        self.determine_older_done_items(recent_date)
+        self.determine_show_all_done()
```
Comment (Jonesey13) [inline todo_app/todo_view_model.py]: These three lines shouldn't be in the constructor (along with the `recent_date` argument). 

We shouldn't be trying to cache any values here. Let them be loaded lazily (like you've done with `to_do_items` & `doing_items` below)

File: todo_app/todo_view_model.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        return self._recent_items
+
+    @property
+    def older_done_items(self):
+        return self._older_items
+
+    def determine_show_all_done(self):
+        done_items = self.done_items
+        if len(done_items) >= 5:
+            self._show_all_done = False
```
Comment (Jonesey13) [inline todo_app/todo_view_model.py]: This property doesn't really belong to this class. I'd just pass it through to the `render_template` method and use it in the template. Should allow you strip out alot of methods here and reduces the number of tests.

**Review by Jonesey13 — COMMENTED** [review]
Left a few comments below

**Comment by maheshpatelhome** [comment]
All resolved

**Pull request closed without merging**